### PR TITLE
feat(ui): Add password change forms + tests

### PR DESCRIPTION
**Pull request by billyvg**
Here's the range that should be reviewed:
https://github.com/getsentry/sentry/pull/7211/files/76d31e5db3952f9e6c838e548ddd7af155e0f421..98bc92db0a0d24f262ee70737039fe1c3011342c

This adds a password change form to "Account Security"
image

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/static/sentry/app/views/settings/components/forms/formField/index.jsx [hunk 1]
```diff
@@ -295,7 +295,7 @@ class FormField extends React.Component {
                         onKeyDown: this.handleKeyDown,
                         onChange: this.handleChange,
                         onBlur: this.handleBlur,
-                        value,
+                        value: value === null ? '' : value,
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/components/forms/formField/index.jsx]: Fixes react warnings about input switching from controlled to uncontrolled

File: src/sentry/static/sentry/app/views/settings/components/forms/jsonForm.jsx [hunk 2]
```diff
@@ -90,6 +92,7 @@ class JsonForm extends React.Component {
                     />
                   );
                 })}
+                {shouldRenderFooter && renderFooter({title, fields})}
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/components/forms/jsonForm.jsx]: Needed an easy way to render a footer when using JsonForm

File: src/sentry/static/sentry/app/views/settings/account/accountSecurity/accountSecurityEnroll.jsx [hunk 3]
```diff
@@ -208,7 +207,7 @@ class AccountSecurityEnroll extends AsyncView {
 
     let data = {
       ...this._form,
-      ...((dataModel && dataModel.toJSON()) || {}),
+      ...(dataModel || {}),
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/views/settings/account/accountSecurity/accountSecurityEnroll.jsx]: why isn't this needed anymore?

File: src/sentry/static/sentry/app/views/settings/components/forms/formField/index.jsx (hunk 1 above)
Comment (MaxBittker) [reply src/sentry/static/sentry/app/views/settings/components/forms/formField/index.jsx]: this helped me understand what was going on, maybe worth including as a comment?

File: src/sentry/static/sentry/app/views/settings/components/forms/model.jsx [hunk 4]
```diff
@@ -167,6 +171,36 @@ class FormModel {
...
+
+  /**
+   * Form data that will be sent to API endpoint (i.e. after transforms)
+   */
+  getTransformedData() {
+    let form = this.getData();
+
+    return Object.keys(form)
+      .map(id => [id, this.getTransformedValue(id)])
+      .reduce((acc, [id, value]) => {
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/views/settings/components/forms/model.jsx]: https://lodash.com/docs/4.17.5#fromPairs  instead of this reduce?

Comment (MaxBittker) [reply src/sentry/static/sentry/app/views/settings/components/forms/model.jsx]:
```
_.fromPairs([['a', 1], ['b', 2]]);
// => { 'a': 1, 'b': 2 }
```

File: src/sentry/static/sentry/app/views/settings/components/forms/model.jsx [hunk 5]
```diff
@@ -233,6 +267,55 @@ class FormModel {
...
+      // Validate required fields
+      ...(Array.from(this.fieldDescriptor.keys()).filter(id => !this.isValidField(id)) ||
+        []),
+    ];
+
+    if (errors.length > 0) return null;
+
+    let saveSnapshot = this.createSnapshot();
+
+    // Check if field needs to handle
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/views/settings/components/forms/model.jsx]: what does this comment mean?

Comment (MaxBittker) [reply src/sentry/static/sentry/app/views/settings/components/forms/model.jsx]: just referring to the fact that fields sometimes transform their data?

**Review by MaxBittker — APPROVED** [review]
lgtm 🔒 

nice work adapting the form to work with this

**Pull request merged**